### PR TITLE
homework 26.2

### PR DESCRIPTION
**Pull request by Alkey**

### CODE REVIEW

**Review by frostpv — APPROVED** [review]

**Comment by frostpv** [comment]
Просмотрел, у меня нет претензий ))) Хорошая работа

File: src/main/java/com/internet/shop/controllers/CompleteOrderController.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private OrderService orderService = (OrderService) injector.getInstance(OrderService.class);
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        orderService.completeOrder(shoppingCartService
+                .getByUserId(Long.parseLong(req.getParameter("shoppingCart"))));
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/CompleteOrderController.java]: let's user constant USER_ID here

File: src/main/java/com/internet/shop/controllers/GetAllOrderProductsController.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.internet.shop.lib.Injector;
+import com.internet.shop.service.OrderService;
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class GetAllOrderProductsController extends HttpServlet {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/GetAllOrderProductsController.java]:
```suggestion
public class GetOrderDetailsController extends HttpServlet {
```

File: src/main/java/com/internet/shop/controllers/GetShoppingCartController.java [hunk 3]
```diff
@@ -17,6 +17,7 @@
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp)
             throws ServletException, IOException {
+        req.setAttribute("shoppingCart", shoppingCartService.getByUserId(USER_ID).getId());
         req.setAttribute("products", shoppingCartService.getByUserId(USER_ID).getProducts());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/GetShoppingCartController.java]: you are calling the same method twice(each time it's a request to DB)

File: src/main/java/com/internet/shop/controllers/CompleteOrderController.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    private ShoppingCartService shoppingCartService =
+            (ShoppingCartService) injector.getInstance(ShoppingCartService.class);
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        orderService.completeOrder(shoppingCartService
+                .getByUserId(USER_ID));
+        resp.sendRedirect("/order/all-orders");
+
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/CompleteOrderController.java:25]:
```suggestion
```

File: src/main/web/WEB-INF/web.xml [hunk 5]
```diff
@@ -74,4 +74,76 @@
...
+        <url-pattern>/shopping-carts/products/delete</url-pattern>
+    </servlet-mapping>
+
+    <servlet>
+        <servlet-name>completeOrder</servlet-name>
+        <servlet-class>com.internet.shop.controllers.CompleteOrderController</servlet-class>
+    </servlet>
+    <servlet-mapping>
+        <servlet-name>completeOrder</servlet-name>
+        <url-pattern>/shopping-carts/products/create-order</url-pattern>
```
Comment (dmytro-hryhoruk) [inline src/main/web/WEB-INF/web.xml:93]:
```suggestion
        <url-pattern>order/create</url-pattern>
```

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**